internal/oobmigration/migrations/batches/role_assignment_migrator.go

```diff
@@ -0,0 +1,82 @@
+package batches
+
+import (
+	"context"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/oobmigration"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+)
+
+type roleAssignmentMigrator struct {
+	store     *basestore.Store
+	batchSize int
+}
+
+func NewRoleAssignmentMigrator(store *basestore.Store, batchSize int) *roleAssignmentMigrator {
+	return &roleAssignmentMigrator{
+		store:     store,
+		batchSize: batchSize,
+	}
+}
+
+var _ oobmigration.Migrator = &roleAssignmentMigrator{}
+
+func (m *roleAssignmentMigrator) ID() int                 { return 19 }
+func (m *roleAssignmentMigrator) Interval() time.Duration { return time.Second * 10 }
+
+// Progress returns the percentage (ranged [0, 1]) of users without a system role (USER or SITE_ADMINISTRATOR) role.
```

Comment:
Based on the tests, I think this is backwards -- it seems that this returns the percentage of users who _do_ have a system role. In the tests, it's 0.0 before the migrator runs, and 1.0 after it runs.

```suggestion
// Progress returns the percentage (ranged [0, 1]) of users who have a system role (USER or SITE_ADMINISTRATOR) assigned.
```

Reply:
This is the correct behavior. The range returned is indicative of the percentage of work completed.

With this migration, we batch users 500 at a time, so before we start, it's at 0% because none of the users have roles assigned as the number of users with assigned roles increases, the progress bar increases. At 100%, every user has a role assigned to them which is on par with the expectation of the OOB migratory service.